hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/NodeDecommissionManager.java

```diff
@@ -368,6 +382,50 @@ public synchronized void startDecommission(DatanodeDetails dn)
     }
   }
 
+  private synchronized boolean checkIfDecommissionPossible(List<DatanodeDetails> dns, List<DatanodeAdminError> errors) {
+    // do we require method synchronization?
+    int minInService = -1; // maxRatis = -1, maxEc = -1;
+    for (DatanodeDetails dn : dns) {
+      Set<ContainerID> containers;
+      try {
+        containers = nodeManager.getContainers(dn);
+      } catch (NodeNotFoundException ex) {
+        LOG.warn("The host {} was not found in SCM. Ignoring the request to " +
+            "decommission it", dn.getHostName());
+        errors.add(new DatanodeAdminError(dn.getHostName(),
+            "The host was not found in SCM"));
+        continue; // ignore the DN and continue to next one
+      }
+      for (ContainerID cid : containers) {
+        ContainerInfo cif;
+        try {
+          cif = containerManager.getContainer(cid);
+        } catch (ContainerNotFoundException ex) {
+          continue; // ignore the container and continue to next one
+        }
+        if (cif.getState().equals(HddsProtos.LifeCycleState.DELETED) ||
+            cif.getState().equals(HddsProtos.LifeCycleState.DELETING)) {
+          continue;
+        }
+        int reqNodes = cif.getReplicationConfig().getRequiredNodes();
+        if (reqNodes > minInService) {
```

Comment:
If you move the line `(nodeManager.getNodeCount(NodeStatus.inServiceHealthy()` to get the total nodes, you don't need to track the minInService. You can just fail early if there are not enough nodes, eg:

```
inService = odeManager.getNodeCount(NodeStatus.inServiceHealthy()) - dns.size();
for (DatanodeDetails dn : dns) {
    ...
    if (cif.getReplicationConfig().getRequiredNodes() > inService) {
        // fail early - no point in checking any further
        return false;
    }
}
```

```